lib/pages/route/route_list.dart

```diff
@@ -85,13 +85,15 @@ class _ListadoRutasScreenState extends State<ListadoRutasScreen> {
         return rutaDificultad == filtroNormalizado;
       }).toList();
     }
-
-    if (_filtroNombre.isNotEmpty) {
-      rutas = rutas?.where((ruta) => ruta['nombre'].toString().toLowerCase().contains(_filtroNombre.toLowerCase())).toList();
+    if (_filtroNombre.isNotEmpty) {rutas = rutas?.where((ruta) => ruta['nombre'].toString().toLowerCase().contains(_filtroNombre.toLowerCase())).toList();
     }
-
     if (_filtroEstrellas != null) {
-      rutas = rutas?.where((ruta) => ruta['estrellas'] == _filtroEstrellas!).toList();
+      rutas = rutas?.where((ruta) {
+        if (ruta['estrellas'] != null) {double estrellasRuta = (ruta['estrellas'] as num).toDouble();
```

Comment:
esto debería estar separado en 2 lineas
```suggestion
        if (ruta['estrellas'] != null) {
          double estrellasRuta = (ruta['estrellas'] as num).toDouble();
```

Reply:
Realizado